src/analyses/base.ml

```diff
@@ -2288,352 +2387,352 @@ struct
-                | _ -> VD.top_value (unrollType dest_typ)
-              end in
-          set ~ctx st dest_a dest_typ value
-        | None -> st
-      end
-    | Strstr { haystack; needle }, _ ->
-      begin match lv with
-        | Some lv_val ->
-          (* check if needle is a substring of haystack in string literals domain if haystack and needle are string literals,
-             else check in null bytes domain if both haystack and needle are / can be transformed to an array domain representation;
-             if needle is substring, assign the substring of haystack starting at the first occurrence of needle to dest,
-             if it surely isn't, assign a null_ptr *)
-          string_manipulation haystack needle lv true (Some (fun h_a n_a -> Address (AD.substring_extraction h_a n_a)))
-            (fun h_ar n_ar -> match CArrays.substring_extraction h_ar n_ar with
-               | CArrays.IsNotSubstr -> Address (AD.null_ptr)
-               | CArrays.IsSubstrAtIndex0 -> Address (eval_lv ~ctx st (mkMem ~addr:(Cil.stripCasts haystack) ~off:NoOffset))
-               | CArrays.IsMaybeSubstr -> Address (AD.join (eval_lv ~ctx st
-                                                              (mkMem ~addr:(Cil.stripCasts haystack) ~off:(Index (Offset.Index.Exp.any, NoOffset)))) (AD.null_ptr)))
-        | None -> st
-      end
-    | Strcmp { s1; s2; n }, _ ->
-      begin match lv with
-        | Some _ ->
-          (* when s1 and s2 are string literals, compare both completely or their first n characters in the string literals domain;
-             else compare them in the null bytes array domain if they are / can be transformed to an array domain representation *)
-          string_manipulation s1 s2 lv false (Some (fun s1_a s2_a -> Int (AD.string_comparison s1_a s2_a (eval_n n))))
-            (fun s1_ar s2_ar -> Int (CArrays.string_comparison s1_ar s2_ar (eval_n n)))
-        | None -> st
-      end
-    | Abort, _ -> raise Deadcode
-    | ThreadExit { ret_val = exp }, _ ->
-      begin match ThreadId.get_current (Analyses.ask_of_ctx ctx) with
-        | `Lifted tid ->
-          (
-            let rv = eval_rv ~ctx ctx.local exp in
-            ctx.sideg (V.thread tid) (G.create_thread rv);
-            (* TODO: emit thread return event so other analyses are aware? *)
-            (* TODO: publish still needed? *)
-            publish_all ctx `Return; (* like normal return *)
-            let ask = Analyses.ask_of_ctx ctx in
-            match ThreadId.get_current ask with
-            | `Lifted tid when ThreadReturn.is_current ask ->
-              ignore @@ Priv.thread_return ask (priv_getg ctx.global) (priv_sideg ctx.sideg) tid st
-            | _ -> ())
-        | _ -> ()
-      end;
-      raise Deadcode
-    | MutexAttrSetType {attr = attr; typ = mtyp}, _ ->
-      begin
-        let get_type lval =
-          let address = eval_lv ~ctx st lval in
-          AD.type_of address
+      | Memset { dest; ch; count; }, _ ->
```

Comment:
These unrelated changes should be undone to minimize the diff.

Reply:
I reverted these changes. However, now the indentation check fails, because of this file.